src/c/tests/TestClient.cc

```diff
@@ -1,1397 +0,0 @@
-        waitForVoidCompletion(3);
-        CPPUNIT_ASSERT(count == 0);
-
-        char buf[1024];
-        int blen = sizeof(buf);
-        struct Stat stat;
-        rc = zoo_get(zk, "/tauth1", 0, buf, &blen, &stat);
-        CPPUNIT_ASSERT_EQUAL((int)ZNOAUTH, rc);
-        // add auth pat w/correct pass verify success
-        rc = zoo_add_auth(zk, "digest", "pat:passwd", 10, voidCompletion,
-                          (void*)ZOK);
-
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-        rc = zoo_get(zk, "/tauth1", 0, buf, &blen, &stat);
-        CPPUNIT_ASSERT_EQUAL((int)ZOK, rc);
-        waitForVoidCompletion(3);
-        CPPUNIT_ASSERT(count == 0);
-        //create a new client
-        zk = createClient(&ctx3);
-        rc = zoo_add_auth(zk, "digest", "pat:passwd", 10, voidCompletion, (void*) ZOK);
-        waitForVoidCompletion(3);
-        CPPUNIT_ASSERT(count == 0);
-        rc = zoo_add_auth(zk, "ip", "none", 4, voidCompletion, (void*)ZOK);
-        //make the server forget the auths
-        waitForVoidCompletion(3);
-        CPPUNIT_ASSERT(count == 0);
-
-        stopServer();
-        CPPUNIT_ASSERT(ctx3.waitForDisconnected(zk));
-        startServer();
-        CPPUNIT_ASSERT(ctx3.waitForConnected(zk));
-        // now try getting the data
-        rc = zoo_get(zk, "/tauth1", 0, buf, &blen, &stat);
-        CPPUNIT_ASSERT_EQUAL((int)ZOK, rc);
-        // also check for get
-        rc = zoo_get_acl(zk, "/", &nodeAcl, &stat);
-        CPPUNIT_ASSERT_EQUAL((int)ZOK, rc);
-        // check if the acl has all the perms
-        CPPUNIT_ASSERT_EQUAL((int)1, (int)nodeAcl.count);
-        acl_val = *(nodeAcl.data);
-        CPPUNIT_ASSERT_EQUAL((int) acl_val.perms, ZOO_PERM_ALL);
-        // verify on root node
-        rc = zoo_set_acl(zk, "/", -1, &ZOO_CREATOR_ALL_ACL);
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-
-        rc = zoo_set_acl(zk, "/", -1, &ZOO_OPEN_ACL_UNSAFE);
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-
-        //[ZOOKEEPER-1108], test that auth info is sent to server, if client is not
-        //connected to server when zoo_add_auth was called.
-        zhandle_t *zk_auth = zookeeper_init(hostPorts, NULL, 10000, 0, NULL, 0);
-        rc = zoo_add_auth(zk_auth, "digest", "pat:passwd", 10, voidCompletion, (void*)ZOK);
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, rc);
-        sleep(2);
-        CPPUNIT_ASSERT(count == 1);
-        count  = 0;
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, zookeeper_close(zk_auth));
-        
-        // [ZOOKEEPER-800] zoo_add_auth should return ZINVALIDSTATE if
-        // the connection is closed. 
-        zhandle_t *zk2 = zookeeper_init(hostPorts, NULL, 10000, 0, NULL, 0);
-        sleep(1);
-        CPPUNIT_ASSERT_EQUAL((int) ZOK, zookeeper_close(zk2));
-        CPPUNIT_ASSERT_EQUAL(0, zoo_state(zk2)); // 0 ==> ZOO_CLOSED_STATE
-        rc = zoo_add_auth(zk2, "digest", "pat:passwd", 10, voidCompletion, (void*)ZOK);
-        CPPUNIT_ASSERT_EQUAL((int) ZINVALIDSTATE, rc);
```

Comment:
### ✅ Getting better: Large Method
testAuth decreases from 93 to 87 lines of code, threshold = 70